File: python/src/mpeg_o/instrument_config.py
"""``InstrumentConfig`` — human-readable acquisition instrument description."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class InstrumentConfig:
    """Instrument configuration block persisted as HDF5 string attributes.

    All fields default to empty strings because the format spec explicitly
    allows every sub-field to be missing (§3 of ``docs/format-spec.md``).
    """

    manufacturer: str = ""
    model: str = ""
    serial_number: str = ""
    source_type: str = ""
    analyzer_type: str = ""
    detector_type: str = ""

File: java/src/main/java/com/dtwthalion/mpgo/InstrumentConfig.java
/*
 * MPEG-O Java Implementation
 * Copyright (C) 2026 DTW-Thalion
 * SPDX-License-Identifier: LGPL-3.0-or-later
 */
package com.dtwthalion.mpgo;

/**
 * Immutable instrument configuration value class.
 *
 * <p>All fields may be empty strings per format-spec §3 (every
 * sub-field is optional).</p>
 *
 * <p><b>API status:</b> Stable.</p>
 *
 * <p><b>Cross-language equivalents:</b> Objective-C
 * {@code MPGOInstrumentConfig}, Python
 * {@code mpeg_o.instrument_config.InstrumentConfig}.</p>
 *
 * @param manufacturer Instrument manufacturer.
 * @param model        Instrument model.
 * @param serialNumber Serial number.
 * @param sourceType   Ionization source type.
 * @param analyzerType Mass analyzer type.
 * @param detectorType Detector type.
 * @since 0.6
 */
public record InstrumentConfig(
    String manufacturer, String model, String serialNumber,
    String sourceType, String analyzerType, String detectorType) {}
